akka-http/src/main/scala/akka/http/scaladsl/server/directives/FileUploadDirectives.scala

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.http.scaladsl.server.directives
+
+import java.io.File
+
+import akka.stream.io.Implicits._
+import akka.stream.scaladsl._
+import akka.http.scaladsl.server.Directive1
+import akka.http.scaladsl.model.{ ContentType, Multipart }
+import scala.util.Try
+
+trait FileUploadDirectives {
+
+  import BasicDirectives._
+  import FutureDirectives._
+  import MarshallingDirectives._
+
+  /**
+   * Collects the bytes of the first file part from the request into the given destination file
+   *
+   * @param destination Will be overwritten if not empty
+   */
+  def uploadedFile(destination: File): Directive1[Try[FormFile]] =
```

Comment:
On one hand I like that, one super simple and one very flexible but on the other hand, that is not very far off from what we already have - basically a filter and a map on the existing stream of body parts? So, bigger api surface for little gain? 

I think the super-simple one is great to keep the threshold for newcomers low and make "the simple things very easy", so we should definitely keep it or something close to it.